EndToEndTests/CustomersAndOrdersEndToEndTest.cs

```diff
@@ -48,13 +48,85 @@ public async Task OrderShouldbeCreated()
                 request.CustomerId = 1;
                 request.OrderTotal = new Money("12.30");
                 var serializeObj = JsonSerializer.Serialize(request);
+                var response = await RestAPICall(uri, serializeObj);
+                Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+                string jsonString = await response.Content.ReadAsStringAsync();
+                var orderResponse = JsonSerializer.Deserialize<CreateOrderResponse>(jsonString, SerializerOptions());
+                Assert.IsNotNull(orderResponse.OrderId);
+            }
+        }
+        [TestMethod]
+        public async Task ShouldApprove()
+        {
+            var customerId = await CreateCustomer("Joe", "50.30");
+            Assert.IsNotNull(customerId);
+            var orderId = await CreateOrder(customerId, "20.30");
+            Assert.IsNotNull(orderId);
+            await AssertOrderState(orderId, OrderState.APPROVED);
+        }
+        [TestMethod]
+        public async Task ShouldReject()
+        {
+            var customerId = await CreateCustomer("Joe", "50.30");
+            Assert.IsNotNull(customerId);
+            var orderId = await CreateOrder(customerId, "120.50");
+            Assert.IsNotNull(orderId);
+            await AssertOrderState(orderId, OrderState.REJECTED);
+        }
+        private async Task<long> CreateCustomer(string name, string amount)
+        {
+            var uri = hostCustmer + "customer";
+            CreateCustomerRequest request = new CreateCustomerRequest();
+            request.Name = name;
+            request.CreditLimit = new Money(amount);
+            var serializeObj = JsonSerializer.Serialize(request);
+            var response = await RestAPICall(uri, serializeObj);
+            string jsonString = await response.Content.ReadAsStringAsync();
+            var customerResponse = JsonSerializer.Deserialize<CreateCustomerResponse>(jsonString, SerializerOptions());
+            return customerResponse.CustomerId;
+        }
+        private async Task<long> CreateOrder(long customerId, string amount)
+        {
+            var uri = hostOrder + "order";
+            CreateOrderRequest request = new CreateOrderRequest();
+            request.CustomerId = customerId;
+            request.OrderTotal = new Money(amount);
+            var serializeObj = JsonSerializer.Serialize(request);
+            var response = await RestAPICall(uri, serializeObj);
+            string jsonString = await response.Content.ReadAsStringAsync();
+            var orderResponse = JsonSerializer.Deserialize<CreateOrderResponse>(jsonString, SerializerOptions());
+            return orderResponse.OrderId;
+        }
+        private async Task AssertOrderState(long orderId, OrderState orderState)
+        {
+            Thread.Sleep(30000); // Wait
```

Comment:
Please do not sleep.
The code should loop for a certain number of iterations.
Each iteration should check that the asserts are successful
If they are it should exit the loop.
Otherwise, it should sleep.
See Eventually.eventually() in the Java version

Reply:
Done